### ISSUE TITLE
The provided regular expression is using multiline anchors (^ or $), which may present a security risk

### ISSUE THREAD

**Issue by fagnerpereira** [webmerge/api-ruby#7]
There is a problem with the regex in Notification class, line 8.
http://homakov.blogspot.co.uk/2012/05/saferweb-injects-in-various-ruby.html
Tks

**Comment by webmerge** [comment]
@fagnerpereira can you submit a pull request with the fix?

**Comment by ottohenrique** [comment]
Hi!

Actually the "bug has already been fixed in this commit https://github.com/webmerge/api-ruby/commit/4b632c156646e39a74efeaf671822a746dec5bd2, but the published version of them gem (1.0.0 on rubygems) is outdated.

Do you have plans or a schedule for a new release?

Thank you for your attention :)

**Comment by webmerge** [comment]
@jheth Can you update the rubygem?  Also, could you transfer ownership of the gem to username: webmerge?

Thanks!!

**Comment by jheth** [comment]
@webmerge  I just pushed version 1.0.1 to rubygems and opened a pull request with the new version number.  I also added you as an owner.  Do you want me to remove myself?

**Comment by webmerge** [comment]
@jheth thanks! You can stay on as owner

**Comment by plicjo** [comment]
This is still an issue in version 1.0.1, even though the code is fixed on master!

I opened up the version 1.0.1 of the gem inside my project and the code still has the old regex:

v1.0.1
```ruby
module WebMerge
  class Notification
    ...
    validates_format_of :to, :from, :with => /^(|(([A-Za-z0-9]+_+)|([A-Za-z0-9]+\-+)|([A-Za-z0-9]+\.+)|([A-Za-z0-9]+\++))*[A-Za-z0-9]+@((\w+\-+)|(\w+\.))*\w{1,63}\.[a-zA-Z]{2,6})$/i
    ...
```

In the next version of the gem, that line should look like it does on master: https://github.com/webmerge/api-ruby/blob/master/lib/web_merge/notification.rb#L8

v 1.0.2?
```ruby
    ...
    validates_format_of :to, :from, :with => /\A(|(([A-Za-z0-9]+_+)|([A-Za-z0-9]+\-+)|([A-Za-z0-9]+\.+)|([A-Za-z0-9]+\++))*[A-Za-z0-9]+@((\w+\-+)|(\w+\.))*\w{1,63}\.[a-zA-Z]{2,6})\z/i
    ...
```

**Comment by plicjo** [comment]
This can be closed, as it's fixed and published in the v1.0.2 of the gem.
